master/lib/Munin/Master/Config.pm

```diff
@@ -470,17 +470,26 @@ sub _split_config_line_ok {
 
 sub _parse_config_line {
     # Parse and save contents of random user configuration.
-    my ($self, $prefix, $key, $value) = @_;
-    
-    my $longkey = $self->_concat_config_line_ok($prefix,$key,$value);
+    my ($self, $prefix, $key, $value, $skip_on_error) = @_;
+    my $longkey;
 
+    if ($skip_on_error) {
+        eval { $longkey = $self->_concat_config_line_ok($prefix, $key, $value); };
+        # Skip single malformed lines (error messages were emitted before).
+        # This is suitable for fault tolerant parsing of "datafile".
+        return if $EVAL_ERROR;
+    } else {
+        # A problem with a malformed line is allowed to rise to the file level.
+        # This is suitable for configuration files.
+        $longkey = $self->_concat_config_line_ok($prefix, $key, $value);
```

Comment:
You could always evaluate, and defer only the error handling depending on the `$skip_on_error`, but I don't think it's worth changing the code, as it will be removed anyway in 3.0.

And, right now, it's dead-simple to read & already working, so let's keep it that way.